dask-gateway-dev.yaml

```
@@ -0,0 +1,58 @@
+# OPTIONAL conda environment file to help get setup with
+# dependencies to build + test dask-gateway locally
+name: dask-gateway-dev
+channels:
+  - conda-forge
+dependencies:
+  # python and pip
+  - python=3.9
```

Let's go for Python 3.10 while we are at it. I've learned that the dask ecosystem is quickly adopting and using modern versions of Python. Currently Python 3.8-3.10 is supported but Python 3.7 that hasn't yet reached end of life is no longer supported.

```suggestion
  - python=3.10
```